### DISCUSSION TITLE
May I ask if SLINT can achieve multiple document interface？

### DISCUSSION THREAD

**Discussion by pcevent** [slint-ui/slint#2923] [Q&A] [answered]
May I ask if SLINT can achieve multiple document interface？

**Comment by ogoffart** [comment]
Could you elaborate what exactly you mean by multiple document interface?

You can always open the main window several time with different document.

If you are talking about subwindows, it is somehow possible but there are still work to do (eg, dynamic z orderning)
[demo](https://slint.dev/editor/preview?load_url=https://raw.githubusercontent.com/slint-ui/slint/master/examples/fancy_demo/main.slint)

**Comment by pcevent** [comment]
At the beginning of the program, multiple windows can be opened using the following code:
```rust
fn main() -> Result<(), slint::PlatformError> {
  let ui = AppWindow::new().unwrap();
  ui.show().unwrap();
  let ui = AppWindow::new().unwrap();
  ui.run()
}
```
I want to display a new window, similar to File Manager (opening from a new window), when I click a button on the button. At this point, no new window is displayed, and the code is as follows:
```rust
    ui.on_button_clicked({
        move || {
          let ui = AppWindow::new().unwrap();
          ui.show().unwrap();
          println!("clicked");
        }
    });
            
```

**Reply by ogoffart** [reply, marked as answer]
The problem is that ui is dropped when it goes out of scope.

You need to store a strong reference of ui somewhere so it doesn't get dropped.

**Comment by pcevent** [comment]
Okay, thank you!
